### PR TITLE
K8SPXC-333 pass extra fields

### PR DESCRIPTION
**Pull request by delgod**
[K8SPXC-333](https://jira.percona.com/browse/K8SPXC-333)

Version Service should know the platform, CR id, and k8s, pmm, backup versions.

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| storage | passed |
| big-data | passed |
| init-deploy | failed |
| recreate | passed |
| self-healing | passed |
| limits | passed |
| scaling-proxysql | passed |
| monitoring | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| monitoring-2-0 | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| security-context | passed |
| operator-self-healing | passed |
| affinity | passed |
| demand-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | failed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

File: pkg/controller/pxc/version.go [hunk 1]
```diff
@@ -69,6 +70,17 @@ func (r *ReconcilePerconaXtraDBCluster) sheduleEnsurePXCVersion(cr *api.PerconaX
 			return
 		}
 
+		sv, err := version.Server()
+		if err != nil {
+			log.Error(err, "get version")
+			return
+		}
+		_, err = localCr.CheckNSetDefaults(sv)
```
Comment (qJkee) [inline pkg/controller/pxc/version.go:78]: You can use `r.ServerVersion` here without calling `version.Server()`

Comment (qJkee) [reply pkg/controller/pxc/version.go:78]: UPDT
`r.serverVersion`

File: pkg/controller/pxc/vs.go [hunk 2]
```diff
@@ -121,3 +154,13 @@ type OperatorVersion struct {
 type VersionResponse struct {
 	Versions []OperatorVersion `json:"versions"`
 }
+
+type currentVersionMeta struct {
```
Comment (blez) [inline pkg/controller/pxc/vs.go:158]: IMHO, `current` is redundant here

Comment (qJkee) [reply pkg/controller/pxc/vs.go:158]: but it clarify what is this version meta about

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| recreate | passed |
| storage | passed |
| scaling | failed |
| big-data | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| limits | passed |
| upgrade | failed |
| monitoring | passed |
| self-healing-advanced | passed |
| restore-to-encrypted-cluster | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| demand-backup | failed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | failed |
| upgrade-consistency | failed |

File: pkg/controller/pxc/vs.go (hunk 2 above)
Comment (delgod) [reply pkg/controller/pxc/vs.go]: fixed in https://github.com/percona/percona-xtradb-cluster-operator/pull/468/commits/3751f45d05c192dfffc1c2b1d4d2241b8ebd3f37

File: pkg/controller/pxc/version.go [hunk 3]
```diff
@@ -98,26 +104,37 @@ func (r *ReconcilePerconaXtraDBCluster) ensurePXCVersion(cr *api.PerconaXtraDBCl
 		return errors.New("cluster is not ready")
 	}
 
-	new, err := vs.GetExactVersion(cr.Spec.UpgradeOptions.Apply, cr.Status.PXC.Version)
+	newVersion, err := vs.GetExactVersion(versionMeta{
+		Apply:         cr.Spec.UpgradeOptions.Apply,
+		Platform:      string(cr.Spec.Platform),
+		KubeVersion:   string(r.serverVersion.Info.GitVersion),
```
Comment (dAdAbird) [inline pkg/controller/pxc/version.go:110]: no need in conversion, these params are already strings

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| init-deploy | passed |
| big-data | passed |
| scaling | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| limits | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| monitoring | failed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| operator-self-healing | passed |
| demand-backup | failed |
| affinity | passed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |

File: pkg/controller/pxc/version.go (hunk 3 above)
Comment (qJkee) [reply pkg/controller/pxc/version.go]: fixed

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| storage | failed |
| scaling | passed |
| init-deploy | passed |
| big-data | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | failed |
| upgrade | passed |
| limits | passed |
| self-healing-advanced | passed |
| demand-backup | failed |
| upgrade-consistency | passed |
| security-context | passed |
| operator-self-healing | passed |
| monitoring | passed |
| scheduled-backup | failed |
| monitoring-2-0 | passed |
| affinity | passed |
| demand-backup-encrypted-with-tls | passed |
| one-pod | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| init-deploy | passed |
| scaling | passed |
| storage | passed |
| self-healing | passed |
| recreate | passed |
| scaling-proxysql | passed |
| limits | passed |
| monitoring | passed |
| upgrade | failed |
| restore-to-encrypted-cluster | passed |
| self-healing-advanced | failed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| affinity | passed |
| one-pod | failed |
| scheduled-backup | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |
| upgrade-consistency | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| big-data | failed |
| scaling | passed |
| recreate | passed |
| storage | passed |
| self-healing | passed |
| limits | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | failed |
| upgrade | passed |
| demand-backup | failed |
| monitoring | passed |
| self-healing-advanced | failed |
| upgrade-consistency | passed |
| security-context | passed |
| monitoring-2-0 | passed |
| scheduled-backup | failed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| demand-backup-encrypted-with-tls | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |

**Pull request merged**